src/main/java/org/folio/dao/impl/FileDefinitionDaoImpl.java

```diff
@@ -0,0 +1,72 @@
+package org.folio.dao.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.sql.UpdateResult;
+import org.folio.dao.FileDefinitionDao;
+import org.folio.rest.jaxrs.model.FileDefinition;
+import org.folio.rest.persist.Criteria.Criteria;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.interfaces.Results;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import java.util.Optional;
+
+@Repository
+public class FileDefinitionDaoImpl implements FileDefinitionDao {
+  private final Logger logger = LoggerFactory.getLogger(FileDefinitionDaoImpl.class);
+  private static final String TABLE = "file_definitions";
+  private static final String ID_FIELD = "'id'";
+
+  private PostgresClientFactory pgClientFactory;
+
+  public FileDefinitionDaoImpl(@Autowired PostgresClientFactory pgClientFactory) {
+    this.pgClientFactory = pgClientFactory;
+  }
+
+  @Override
+  public Future<Optional<FileDefinition>> getById(String id, String tenantId) {
+    Promise<Results<FileDefinition>> promise = Promise.promise();
+    try {
+      Criteria idCrit = constructCriteria(ID_FIELD, id);
+      pgClientFactory.getInstance(tenantId).get(TABLE, FileDefinition.class, new Criterion(idCrit), false, promise);
```

Comment:
can we use PgUtil? All this exception handling is already done there, all this ocde will be reduced to a single line:
```suggestion
       PgUtil.getById(TABLE, FileDefinition.class, id, okapiHeaders, vertxContext, GetDataExportFileDefinitionsByFileDefinitionIdResponse.class, asyncResultHandler);
```

Reply:
You need not return an Optional in that case, the method will directly throw a 404

Reply:
Let's do not use PgUtil at all. PgUtil does not useful enough to work with the database, because it is API is tightly coupled to the Response and forces DAO to form responses:)   Data access layer should be responsible only for data retrieval.